icu4c/source/i18n/unicode/numberformatter.h

```diff
@@ -2041,6 +2046,49 @@ class U_I18N_API NumberFormatterSettings {
      */
     Derived scale(const Scale &scale) &&;
 
+    /**
+     * Specifies the usage of the unit ("person", "road", "person" ...etc.)
+     *
+     * <p>
+     * NOTE: `usage` will change the output unit depending on the `Locale`  
+     * and the unit value.
+     *    For Example:
+     *        Locale: en_US
+     *        Usage : length-person
+     *        Unit  : Meter
+     *
+     *        If the unit value is 0.25, the output will be "10 inches."
+     *        If the unit value is 1.50, the output will be 
+     *                                           "4 feet and 11 inches"
+     *</p>
+     *
+     * <P>
+     * If the input usage is not exist (e.g. "dance") or is misspelled,
+     * the usage will be **fall backed** to the "default" usage.
+     * </p>
+     *
+     * Pass this method a `StringPiece` that represents the usage of 
+     * the unit. For example:
+     *
+     * <pre>
+     * NumberFormatter::with().usage("person")
+     * </pre>
+     *
+     * @param usage The unit `usage`.
```

Comment:
I think we should allow zero-length strings to unset the usage.  In other words, the default usage is the empty string, which never causes a unit conversion to take place, and any string with nonzero length triggers the fallback to default if it's not available.

Thoughts?

Reply:
We have two special usage parameters:

1. use the output units that are customary for the locale in the absence of any specific usage. Also use if the input usage value is not supported. In CLDR, usage="default". Eg US + 3 meters => "9.8 feet"

2. don't use the usage parameter at all, just use exactly the input unit. No relevant attribute value in CLDR. Eg US + 3 meters => "3 meters". Used mostly to override a previously set usage parameter.

Assuming that the parameter for # 1 is "default", I'm happy with either "" or null for # 2. So if you prefer "", that sounds great.

Reply:
It feeds into the StringPiece constructor, in which both `nullptr` and `""` are considered empty strings.

Reply:
Ping @younies: please add docs that an empty StringPiece (including nullptr) causes the usage to be unset (Mark's case 2).

Reply:
Since it's a StringPiece and not a StringPiece*, can the usage() function receive a null pointer? I think not? (Is mentioning the behaviour of the StringPiece(nullptr) constructor here necessary/useful?)